File: Linear search/linear_search.java
import java.util.*;
class Main{
    public static void main(String args[]){
        int n,x,result;
        Scanner s= new Scanner(System.in);
        System.out.println("enter the number");
        n=s.nextInt();
        int a[]=new int[n];
        System.out.println("enter the elements");
        for(int i=0;i<n;i++){
            a[i]=s.nextInt();
        }
        System.out.println("enter the element to be searched");
        x=s.nextInt();
        result = search(a, n, x);
        if(result==-1)
            System.out.println("element not found");
        else
            System.out.println("element found at index "+result);

    }

    public static int search(int a[],int n,int x){
        for(int i=0;i<n;i++){
            if(a[i]==x){
                return i;
            }
            
        }
        return -1;
    }
}

File: Linear search/linear_search.c
#include<stdio.h>
int search(int a[],int n,int x){
    for(int i=0;i<n;i++){
        if(a[i]==x){
            return i;
            break;
        }
    }
    return -1;
}

int main(){
    int n,x,result;
    printf("enter the number\n");
    scanf("%d",&n);
    int a[n];
    printf("enter the elements\n");
    for(int i=0;i<n;i++){
        scanf("%d",&a[i]);
    }
    printf("enter the number to be searched: ");
    scanf("%d",&x);
    result = search(a,n,x);
    (result==-1)?printf("element does not found"):printf("element found at index %d",result);
    return 0;
}

